data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/AggregateEventHandle.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.event;
+
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import java.lang.ref.WeakReference;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.BiConsumer;
+import java.time.Instant;
+import java.io.Serializable;
+
+public class AggregateEventHandle implements EventHandle, InternalEventHandle, Serializable {
```

Comment:
Should this class be made internal? Are we going to have processors use this?

Reply:
No. It will be used by AggregateProcessor.